src/extract.jl

```diff
@@ -145,6 +143,28 @@ function git_object_hash(
     return bytes2hex(SHA.digest!(ctx))
 end
 
+function git_file_hash(
+    tar::IO,
+    size::Int,
+    HashType::DataType;
+    buf::Vector{UInt8} = Vector{UInt8}(undef, DEFAULT_BUFFER_SIZE),
+)
+    ctx = HashType()
+    SHA.update!(ctx, codeunits("blob $size\0"))
+    # TODO: this largely duplicates the logic of read_data
+    # read_data could be used directly if SHA offered an interface
+    # where you write data to an IO object and it maintains a hash
+    while size > 0
+        r = readbytes!(tar, buf, size < sizeof(buf) ? round_up(size) : sizeof(buf))
+        r < 512 && eof(io) && error("premature end of tar file")
```

Comment:
This 512 doesn't make a whole lot of sense. Missed when making the `buf` size configurable?

Reply:
@keno: was this missed or is it intentional? Note: copied from your `read_data` method.

Reply:
I guess the more general version here seems like we should compute how much we're looking to read and if we're able to read less than that it's a premature end of file. Testing error conditions is such a pain, I really don't want to...

Reply:
Tar is written 512 byte sectors, so if a tar file isn't a multiple of 512 it's corrupt. Should probably check the size mod 512 though

Reply:
Potential fix: https://github.com/JuliaIO/Tar.jl/pull/40